lib/hmmlearn/hmm.py

```diff
@@ -1049,3 +1050,148 @@ def outer_f(x):  # Outer product over features.
+                 n_iter=10, tol=1e-2, verbose=False,
+                 params="stl", init_params="stl",
+                 implementation="log"):
+        """
+        Parameters
+        ----------
+        n_components : int
+            Number of states.
+
+        lambda_prior : float, array, shape (n_components, n_features), optional
+            The prior on the lambda values. Either an array with lambda
+            values for each component and feature or a lambda value with which
+            to generate random values from a Poisson distribution.
+
+        startprob_prior : array, shape (n_components, ), optional
+            Parameters of the Dirichlet prior distribution for
+            :attr:`startprob_`.
+
+        transmat_prior : array, shape (n_components, n_components), optional
+            Parameters of the Dirichlet prior distribution for each row
+            of the transition probabilities :attr:`transmat_`.
+
+        algorithm : {"viterbi", "map"}, optional
+            Decoder algorithm.
+
+        random_state: RandomState or an int seed, optional
+            A random number generator instance.
+
+        n_iter : int, optional
+            Maximum number of iterations to perform.
+
+        tol : float, optional
+            Convergence threshold. EM will stop if the gain in log-likelihood
+            is below this value.
+
+        verbose : bool, optional
+            Whether per-iteration convergence reports are printed to
+            :data:`sys.stderr`.  Convergence can also be diagnosed using the
+            :attr:`monitor_` attribute.
+
+        params, init_params : string, optional
+            The parameters that get updated during (``params``) or initialized
+            before (``init_params``) the training.  Can contain any
+            combination of 's' for startprob, 't' for transmat, and 'e' for
+            emissionprob.  Defaults to all parameters.
+
+        implementation: string, optional
+            Determines if the forward-backward algorithm is implemented with
+            logarithms ("log"), or using scaling ("scaling").  The default is
+            to use logarithms for backwards compatability.
+        """
+        BaseHMM.__init__(self, n_components,
+                         startprob_prior=startprob_prior,
+                         transmat_prior=transmat_prior,
+                         algorithm=algorithm,
+                         random_state=random_state,
+                         n_iter=n_iter, tol=tol, verbose=verbose,
+                         params=params, init_params=init_params,
+                         implementation=implementation)
+        self.lambda_prior = lambda_prior
+
+    def _init(self, X):
+        _check_and_set_n_features(self, X)
+        super()._init(X)
+        self.random_state = check_random_state(self.random_state)
+
+        if self._needs_init('l', 'lambdas_'):
+            self.lambdas_ = self.random_state.gamma(
```

Comment:
Is there a reason for a gamma distribution here?

Reply:
It just seemed like a reasonable prior for Poisson `lambda` parameters. I'm not sure what other people have used. It seems to work fairly well for a somewhat broad range of data, potentially if `lamdas` are very large, it could be a poor choice but I think that's fairly rare and it's just the starting point.

Reply:
Would it be perhaps slightly more robust to scale the initial guess e.g. to the inverse of the mean of the observations?

Reply:
That's a great idea! It does look like gamma is the conjugate prior for poisson distribution so I think this is the best option, I just forgot why :) https://vioshyvo.github.io/Bayesian_inference/conjugate-distributions.html